### ISSUE TITLE
Release job is broken on out-of-date publish plugin

### ISSUE THREAD

**Issue by TimvdLippe** [mockito/mockito#3235]
We are using Shipkit to automatically publish to Maven Central. This job is currently failing (example https://github.com/mockito/mockito/actions/runs/7519868803/job/20471236051) with:

```
> Staging repository is not in desired state [released, not_found]: StagingRepository(id=orgmockito-1112, state=closed, transitioning=false). It is unexpected. Please check the Nexus logs using its web interface - it can be caused by validation rules violation (e.g. publishing artifacts with the same version again). If not, please report it to https://github.com/gradle-nexus/publish-plugin/issues/ with the '--info' logs.
```

This is supposedly fixed in the latest version of the publish plugin. For that, we need Shipkit to update its version or workaround the older version of Shipkit: https://github.com/shipkit/shipkit-changelog/issues/146

**Comment by TimvdLippe** [comment]
I deleted the 5.9.0 release thinking it was broken, as it didn't work before. However, I just realized that https://search.maven.org/artifact/org.mockito/mockito-core/5.9.0/ exists on maven-central after all. Therefore, I believe that the release was fixed later by Sonatype, as it was crashing on our side.

In any case, the Shipkit dependencies have also been fixed so that this should not reoccur. I will schedule the 5.10.0 release now to test that out.

**Comment by TimvdLippe** [comment]
Okay, messed things up with the tags. I had deleted the 5.9.0 tag thinking it was broken. Then I added a new release + tag, but not on the correct commit according to what is on Maven Central. So I deleted the 5.9.0 release and tag yet again, updated it to the original correct commit. And then I triggered a new 5.10.0 release. Let's hope that one gets through.

**Comment by TimvdLippe** [comment]
Aha. The reason I was confused is that https://repo1.maven.org/maven2/org/mockito/mockito-core/ does **not** list 5.9.0, but if you go directly to https://repo1.maven.org/maven2/org/mockito/mockito-core/5.9.0/ then it does show up.

![image](https://github.com/mockito/mockito/assets/5948271/40f946b2-c2eb-4ab0-8df8-ae92b2f89140)

**Comment by TimvdLippe** [comment]
https://github.com/mockito/mockito/actions/runs/7644885695/job/20830690407 just finished and should have published 5.10.0 as usual. Will monitor https://repo1.maven.org/maven2/org/mockito/mockito-core/5.10.0/ for that.

**Comment by TimvdLippe** [comment]
5.10.0 is available on maven-central. The overview page still doesn't show it, but I guess that's a separate bug. Marking this issue as resolved now.
